### ISSUE TITLE
Netscaler - Ansible to bind/unbind content switching policies

### ISSUE THREAD

**Issue by murilo-rocha** [citrix/citrix-adc-ansible-modules#95]
Hi All,
I am pretty new to Ansible. My background is network and security and I am working to learn more about network automation using ansible.

I have a CS Virtual server and every now and then when there is a need to invoke a maintenance page using the netscaler due to work being done on the backend servers.  To invoke the maintenance page I need to bind a CS responder policy to the CS virtual server.  

There are four existing policies with priorities 90 to 120. Sometimes the policy to invoke the maintenance page is bound with priority 115 and sometimes 125 depending on what is needed.

I am trying to automate that so when there is a need to put up/remove the maintenance page at 1am for instance anyone can do it without having to rely on the network people.

Is it possible to achieve that using an ansible playbook?
Basically I need to be able to work on a playbook to bind the "maintenance page" policy with a specific priority and then another playbook to unbind the maintenance page policy after work on the backend is completed.

Thanks
Murilo

**Comment by giorgos-nikolopoulos** [comment]
The way you would probably go about it is that you would have a playbook defining the cs vserver and its cs policy bindings like so

```
- name: Setup content switching vserver
  delegate_to: localhost
  citrix_adc_cs_vserver:
    nsip: 172.18.0.2
    nitro_user: nsroot
    nitro_pass: nsroot

    state: present

    name: cs_vserver_1
    ipv46: 192.168.1.1
    port: 80
    servicetype: HTTP

    policybindings:
      - policyname: regular_policy_1
        priority: 90
      - policyname: regular_policy_2
        priority: 100
      - policyname: regular_policy_3
        priority: 120

```

When you need to apply the maintenance page you would run to apply the following playbook
```
- name: Setup content switching vserver
  delegate_to: localhost
  citrix_adc_cs_vserver:
    nsip: 172.18.0.2
    nitro_user: nsroot
    nitro_pass: nsroot

    state: present

    name: cs_vserver_1
    ipv46: 192.168.1.1
    port: 80
    servicetype: HTTP

    policybindings:
      - policyname: regular_policy_1
        priority: 90
      - policyname: regular_policy_2
        priority: 100
      - policyname: maintenance_policy
        priority: 115
      - policyname: regular_policy_3
        priority: 120

```
The cs vserver module should take care of the deletion and addition of bindings.

You need to apply the whole configuration again for the cs vserver since ansible has no notion of applying a diff to an existing configuration.

**Comment by murilo-rocha** [comment]
Awesome. Thanks a lot Giorgos.
I will run a lab. It's a pretty complex setup and very critical with cache, rewrite, cs policies etc. so I will see how it goes on a lab.
